bin/varnishncsa/varnishncsa.c

```diff
@@ -786,15 +794,24 @@ dispatch_f(struct VSL_data *vsl, struct VSL_transaction * const pt[],
 	unsigned tag;
 	const char *b, *e, *p;
 	struct watch *w;
-	int i, skip;
+	int i, skip, be_mark;
 	(void)vsl;
 	(void)priv;
 
+#define BACKEND_MARKER VSL_t__MAX
+
 	for (t = pt[0]; t != NULL; t = *++pt) {
 		CTX.gen++;
-		if (t->type != VSL_t_req)
-			/* Only look at client requests */
-			continue;
+
+		/* Consider client requests only if in client mode.
+		   Consider backend requests only if in backend mode. */
+		if (t->type == VSL_t_req && CTX.c_opt) {
+			CTX.side = "client";
+			be_mark = 0;
+		} else if (t->type == VSL_t_bereq && CTX.b_opt) {
+			CTX.side = "backend";
+			be_mark = BACKEND_MARKER;
+		} else continue;
```

Comment:
> In varnishncsa, the variable CTX already had an a_opt member, so for the two other options, b_opt and c_opt was chosen.

Not sure what you meant.

> does not work when both b_opt and c_opt are set. Also, similar constructions would remove safety against "impossible" combinations like SLT_PipeAcct when t->type == VSL_t_bereq . In the current version this bad behavior from varnish would be ignored, and I want to keep the current behavior in the new version.

Ugh? How come it will not work?
I don't think the impossible scenarios justify the extra code. If you are concerned about SLT_PipeAcct you can check that particular one.  We have different records (e.g. ReqHeader vs BereqHeader) for that specific purpose.
The code does not handle backend records in client requests either.